js/components/urlBar.js

```diff
@@ -176,9 +177,20 @@ class UrlBar extends ImmutableComponent {
             // load the selected suggestion
             this.urlBarSuggestions.clickSelected(e)
           } else {
+            if (this.activateSearchEngine && this.searchSelectEntry !== null && !isLocationUrl) {
+              const replaceRE = new RegExp('^' + this.searchSelectEntry.shortcut + ' ', 'g')
```

Comment:
Maybe we should pull `removeEngineShortcut` into utility function in the future, @bsclifton ?